lib/base-compiler.js

```diff
@@ -55,6 +56,11 @@ Compile.prototype.newTempDir = function () {
 Compile.prototype.writeFile = Promise.denodeify(fs.writeFile);
 Compile.prototype.readFile = Promise.denodeify(fs.readFile);
 Compile.prototype.stat = Promise.denodeify(fs.stat);
+Compile.prototype.couldHaveOptOutput = function(options, version) {
+    //TODO(jared): a compiler meta data gatherer could improve this
```

Comment:
Right - this is definitely something on my hit-list: having more information about each compiler without resorting to these kinds of things.

I'm happy for this to start with, but probably a more reliable way is to add a flag to the compiler (defaulted to false, and made true in all the configurations that are `clang`y. e.g. look at how `options` and similar are specified for all clang-like compilers in https://github.com/mattgodbolt/compiler-explorer/blob/master/etc/config/c%2B%2B.amazon.properties#L67